formats_test.go

```diff
@@ -0,0 +1,182 @@
+		fnFormat, timeout, idleTimeout, s.RouteConfig, s.RouteHeaders)
+
+	for i := 0; i < times; i++ {
+		requestBody := fmt.Sprintf(`{"name":"%v"}`, RandStringBytes(100))
+		output, response, err := doRequest(t, s.AppName, fnRoute, "text/plain", requestBody)
+		if err != nil {
+			t.Errorf("Got unexpected error: %v", err)
+		}
+		if response.StatusCode != http.StatusOK {
+			if response.StatusCode == http.StatusNotImplemented {
+				// FDK should respond with HTTP 501 Not Implemented if specified protocol not implemented in FDK
+				t.Logf("It seems like given FDK does not support '%v' format. "+
+					"That's not a reason to fail, continue testing...", fnFormat)
+			} else {
+				t.Log(output.String())
+				t.Errorf("Status code assertion error.\n\tExpected: %v\n\tActual: %v",
+					200, response.StatusCode)
+			}
+		}
+	}
+
+	DeleteApp(t, s.Context, s.Client, s.AppName)
+}
+
+func callOnce(t *testing.T, s *SuiteSetup, fnRoute, fnImage,
+	fnFormat string, requestBody interface{}) (*bytes.Buffer, *http.Response, error) {
+
+	timeout := int32(30)
+	idleTimeout := int32(10)
+	CreateApp(t, s.Context, s.Client, s.AppName, map[string]string{})
+	CreateRoute(t, s.Context, s.Client, s.AppName, fnRoute, fnImage, "sync",
+		fnFormat, timeout, idleTimeout, s.RouteConfig, s.RouteHeaders)
+
+	output, response, err := doRequest(t, s.AppName, fnRoute, "application/json", requestBody)
+	if err != nil {
+		return nil, response, err
+	}
+
+	DeleteApp(t, s.Context, s.Client, s.AppName)
+
+	return output, response, nil
+}
+
+func TestFDKFormatSmallBody(t *testing.T) {
+
+	FDKImage := os.Getenv("FDK_FUNCTION_IMAGE")
+	if FDKImage == "" {
+		t.Error("Please set FDK-based function image to test")
+	}
+	formats := []string{"http", "json"}
+
+	helloJohnPayload := &struct {
+		Name string `json:"name"`
+	}{}
+	ExpectedOutput := "Hello %v"
+	for _, format := range formats {
+
+		// echo function:
+		// payload:
+		//    {
+		//        "name": "%v"
+		//    }
+		// response:
+		//    "Hello %v"
+		// if name is empty then:
+		//	  "Hello World"
+		t.Run(fmt.Sprintf("test-fdk-%v-small-body", format), func(t *testing.T) {
+
+			t.Parallel()
+			s := SetupDefaultSuite()
+			route := fmt.Sprintf("/test-fdk-%v-format-small-body", format)
+
+			for _, part := range []string{"", "Jimmy", RandStringBytes(10), RandStringBytes(20)} {
+				helloJohnPayload.Name = part
+				output, response, err := callOnce(t, s, route, FDKImage, format, helloJohnPayload)
+
+				if err != nil {
+					t.Errorf("unexpected error: %v", err.Error())
+				}
+
+				expected := ""
+				if part == "" {
+					expected = fmt.Sprintf(ExpectedOutput, "World")
+				}
+
+				expected = fmt.Sprintf(ExpectedOutput, part)
+				if !strings.Contains(output.String(), expected) {
+					t.Errorf("Output assertion error.\n\tExpected: %v\n\tActual: %v", expected, output.String())
+				}
+				if response.StatusCode != http.StatusOK {
+					if response.StatusCode == http.StatusNotImplemented {
```

Comment:
It's not clear how an FDK that doesn't understand a specific format can indicate that via a 501 right now (as the status code has to be encoded in the specified format)

Reply:
Would it be better to let the tested formats be specified via env? : 
e.g. : 
```
// FilterTestedFormats filters a list of formats to those that are tested for this FDK, based on a comma-separated list in the "FDK_FORMATS" environment variable
func FilterTestedFormats(formats []string )[]string {
	supportedFormats := os.Getenv("FDK_FORMATS")

	if supportedFormats != "" {
		acceptedFormats := strings.Split(supportedFormats,",")
		validFormats := []string{}

		for _,af := range acceptedFormats {
			for _, reqF := range formats {
				if reqF == af {
					validFormats = append(validFormats,reqF)
				}
			}
		}
		return validFormats

	}
	return formats
}


       formats := FilterTestedFormats([]string{"http", "json"})

	helloJohnPayload := &struct {
		Name string `json:"name"`
	}{}
       ... 
```

Reply:
Okay, minor thing, but don't have objections.